Library/Homebrew/cask/artifact/binary.rb

```diff
@@ -3,14 +3,68 @@
 
 require "cask/artifact/symlinked"
 
+require "extend/hash_validator"
+using HashValidator
+
 module Cask
   module Artifact
     # Artifact corresponding to the `binary` stanza.
     #
     # @api private
     class Binary < Symlinked
-      def link(command: nil, **options)
-        super(command: command, **options)
+      def self.from_args(cask, *args)
+        source, args = args
+
+        if args
+          raise CaskInvalidError unless args.respond_to?(:keys)
+
+          args.assert_valid_keys!(:target, :shimscript)
+        end
+
+        args ||= {}
+
+        new(cask, source, **args)
+      end
+
+      def initialize(cask, source, target: nil, shimscript: nil)
+        super(cask, source, target: target)
+        return if shimscript.nil?
+
+        @shimscript_source = @source
```

Comment:
```suggestion
        @shimscript_source = source
```

Reply:
This has to be `@source`, `self.source`, or `cask.staged_path.join(source) `

Here is where `@source` is set: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/artifact/relocated.rb#L53-L54